src/elements/content-sidebar/versions/VersionsSidebarAPI.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * @flow
+ * @file Versions Sidebar API Helper
+ * @author Box
+ */
+import API from '../../../api';
+import { FILE_VERSION_FIELDS_TO_FETCH } from '../../../utils/fields';
+
+export type fetchPayload = [BoxItem, FileVersions];
+
+export default class VersionsSidebarAPI {
+    api: API;
+
+    fileId: string;
+
+    constructor({ api, fileId }: { api: API, fileId: string }) {
+        this.api = api;
+        this.fileId = fileId;
+    }
+
+    fetchData = (): Promise<fetchPayload> => {
+        return Promise.all([this.fetchFile(), this.fetchVersions()]).then(this.fetchVersionCurrent);
+    };
+
+    fetchDownloadUrl = (version: ?BoxItemVersion): Promise<string> => {
+        return new Promise((resolve, reject) => {
+            if (!version) {
+                return reject(new Error('Could not find requested version'));
+            }
+
+            return this.api.getFileAPI().getDownloadUrl(this.fileId, version, resolve, reject);
+        });
+    };
+
+    fetchFile = (): Promise<BoxItem> => {
+        return new Promise((resolve, reject) =>
+            this.api.getFileAPI().getFile(this.fileId, resolve, reject, {
```

Comment:
Should we update `getFileAPI` to also take the `shouldDestroy` param? As is, would this potentially cause any issues like we saw previously where promises would get unresolved?

Reply:
Possibly, but that seems a bit beyond the scope of this PR.

Reply:
Yeah definitely 👍